src/database/bplus/mod.rs

```diff
@@ -0,0 +1,102 @@
+//! # Bplus Database [WIP]
+//!
+//! This module provides a database implementation backed by a
+//! persistent bplus tree.
+//!
+//! #### Authorship
+//!
+//! - Benjamin Riley Zimmerman, 9/11/2024 (bz931@berkeley.edu)
+
+/* IMPORTS */
+
+use std::path::Path;
+
+use anyhow::Result;
+use bitvec::prelude::{BitSlice, Msb0};
+
+#[allow(unused_imports)]
+use crate::{
+    model::State,
+    database::{Persistence, KVStore, Persistent, Tabular, Record, Schema, SchemaBuilder},
+};
+
+/* CONSTANTS */
+
+
+
+/* DATABASE DEFINITION */
+
+#[allow(dead_code)]
+pub struct Database<'a> {
+    data: &'a str,
+}
+
+#[allow(dead_code)]
+pub struct Parameters<'a> {
+    persistence: Persistence<'a>,
+}
+
+/* IMPLEMENTATION */
+
+#[allow(unused)]
+impl Database<'_> {
+    pub fn initialize(params: Parameters) -> Result<Self> {
+        todo!()
+    }
+}
+
+#[allow(unused)]
+impl<R: Record> KVStore<R> for Database<'_> {
+    fn put(&mut self, key: State, value: &R) {
+        todo!()
+    }
+
+    fn get(&self, key: State) -> Option<&BitSlice<u8, Msb0>> {
+        todo!()
+    }
+
+    fn del(&self, key: State) {
+        todo!()
+    }
+}
+
+#[allow(unused)]
+impl Persistent for Database<'_> {
+    fn bind_path(&self, path: &Path) -> Result<()> {
+        todo!()
+    }
+
+    fn materialize(&self) -> Result<()> {
+        todo!()
+    }
+}
+
+#[allow(unused)]
+impl Tabular for Database<'_> {
+    fn create_table(&self, id: &str, schema: Schema) -> Result<()> {
+        todo!()
+    }
+
+    fn select_table(&self, id: &str) -> Result<()> {
+        todo!()
+    }
+
+    fn delete_table(&self, id: &str) -> Result<()> {
+        todo!()
+    }
+}
+
+impl Record for Database<'_> {
```

Comment:
The database will store records, but it is not a record.

I guess there is some weird sense in which it is, but there is no need for this implementation.

Reply:
sg, removed